Sources/Redis/Client/RedisClient+Channel.swift

```diff
@@ -1,24 +1,41 @@
 import Async
 import Bits
+import NIO
 
 extension RedisClient {
-    /// Creates a `RedisChannelStream`.
-    public static func subscribe<Stream>(to channels: Set<String>, stream: Stream, on worker: Worker) -> Future<RedisChannelStream>
-        where Stream: ByteStream
-    {
-        let client = RedisClient(stream: stream, on: worker)
-        let channels = channels.map { name in
-            return RedisData(bulk: name)
-        }
-        return client.command("SUBSCRIBE", channels).map(to: RedisChannelStream.self) { data in
-            return .init(source: stream, worker: worker)
-        }
+    /// Publishes data to a Redis channel.
+    public func publish(_ message: RedisData, to channel: String) -> Future<RedisData> {
+        return command("PUBLISH", [.bulkString(channel), message])
     }
 }
 
 extension RedisClient {
-    /// Publishes data to a Redis channel.
-    public func publish(_ message: RedisData, to channel: String) -> Future<RedisData> {
-        return command("PUBLISH", [.bulkString(channel), message])
+    /// Subscribes to the channels and call the subscription handler on message from server
+    /// Should be closed when subscription not needed.
+    /// Makes the redis client subscription only.
+    public func subscribe(_ channels: Set<String>,
+                          subscriptionHandler: @escaping (RedisChannelData) -> Void) throws -> Future<Void> {
```

Comment:
all Xcode-aided formatting must be abolished from this PR

Reply:
I am afraid this was an artifact of SwiftLint formatting.  I will go ahead and put all the function bodies in a  single line. to remove the formatting.